proposed/http-middleware/middleware-meta.md

```diff
@@ -1,49 +1,56 @@
-PSR-15 Meta Document
-====================
+HTTP Server Middleware
+======================
 
 1. Summary
 ----------
 
 The purpose of this PSR is to provide an interface that defines the formal
-method signature for HTTP Middleware that is compatible with HTTP Messages,
-as defined in PSR-7.
+method signature for HTTP Server Middleware that is compatible with HTTP
+Messages, as defined in [PSR-7][psr7].
+
+[psr7]: http://www.php-fig.org/psr/psr-7/
 
 2. Why Bother?
 --------------
 
-The general concept of reusable middleware was popularized by [StackPHP][stackphp].
-Since the release of the HTTP Messages standard, a number of frameworks have
-adopted middleware that uses HTTP Message interfaces.
+The HTTP Messages specification does not contain any reference to HTTP Middleware.
+
+The design pattern used by middleware has existed for many years as the
+[pipeline pattern][pipeline], or more specifically, "linear pipeline processing".
```

Comment:
> The approach taken in this proposal is more along a _functional_ paradigm
> …
> This is also why usage of interfaces versus the looser `callable` is preferred, as it does not rely on a _convention_.

@weierophinney I agree on everything, especially on the two statements above. But, because it is more along the _functional_ paradigm, the next natural choice is `__invoke`, and not `process`. `__invoke` shares the same properties as `process`, and allows functional and OOP style:

```php
class M implements MiddlewareInterface
{
    public function process(ServerRequestInterface $request, DelegateInterface $delegate)
    {
        // functional style
        return $delegate($request);
        // OOP style
        return $delegate->__invoke($request);
    }
}
```